benchmark/httpapi.py

```diff
@@ -163,6 +187,56 @@ def delete(self, request, id):
         request.setResponseCode(NO_CONTENT)
         return self.backend.delete(id)
 
+    @app.route("/benchmark-results", methods=['GET'])
+    def query(self, request):
+        """
+        Query the previously stored benchmarking results.
+
+        Currently this method only supports filtering the results by the
+        branch name.
+        There is no support for paging of results, but a limit on the
+        number of the results to return may be specified.
+        The returned results are ordered by the timestamp in descending
+        order.
+
+        :param twisted.web.http.Request request: The request.
+        """
+        request.setHeader(b'content-type', b'application/json')
+        params = self._parse_query_args(request.args)
+        d = self.backend.query(**params)
+
+        def got_results(results):
+            result = {"version": self.version, "results": results}
+            return dumps(result)
+
+        d.addCallback(got_results)
+        return d
+
+    @staticmethod
+    def _parse_query_args(args):
```

Comment:
:grey_question: I dunno if you've considered this (maybe it's too much overhead for this project), but we use https://python-jsonschema.readthedocs.org/en/latest/ in flocker, and it saves us from having to manually do verification like this all over the place. At this point manual verification is probably fine, but having a structured system for it is nice. If you find yourself writing too many more blocks like this we should probably switch to something like that.